crates/packet_inspector/src/context.rs

```diff
@@ -0,0 +1,365 @@
+use std::path::PathBuf;
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::RwLock;
+
+use owo_colors::{OwoColorize, Style};
+use regex::Regex;
+use time::OffsetDateTime;
+use valence_protocol::codec::PacketDecoder;
+use valence_protocol::packet::c2s::handshake::HandshakeC2s;
+use valence_protocol::packet::c2s::login::{LoginHelloC2s, LoginKeyC2s};
+use valence_protocol::packet::c2s::status::{QueryPingC2s, QueryRequestC2s};
+use valence_protocol::packet::s2c::login::LoginSuccessS2c;
+use valence_protocol::packet::s2c::status::{QueryPongS2c, QueryResponseS2c};
+use valence_protocol::packet::{C2sPlayPacket, S2cLoginPacket, S2cPlayPacket};
+
+use crate::packet_widget::{systemtime_strftime, PacketDirection};
+
+#[derive(Clone)]
+pub enum Stage {
+    HandshakeC2s,
+    QueryRequestC2s,
+    QueryResponseS2c,
+    QueryPingC2s,
+    QueryPongS2c,
+    LoginHelloC2s,
+    S2cLoginPacket,
+    LoginKeyC2s,
+    LoginSuccessS2c,
+    C2sPlayPacket,
+    S2cPlayPacket,
+}
+
+#[derive(Clone)]
+pub struct Packet {
+    pub(crate) id: usize,
+    pub(crate) direction: PacketDirection,
+    pub(crate) selected: bool,
+    pub(crate) use_compression: bool,
+    pub(crate) packet_data: Vec<u8>,
+    pub(crate) stage: Stage,
+    pub(crate) packet_type: i32,
+    pub(crate) packet_name: String,
+    pub(crate) created_at: OffsetDateTime,
+}
+
+impl From<&mut Packet> for String {
+    fn from(value: &mut Packet) -> Self {
+        if value.packet_data.len() > 1024 {
+            return "Packet too large".to_string();
```

Comment:
Yeah I'll remove this entirely, it doesnt even work correctly since the length it returns is of the compressed packet. (Which even for chunk data, still fits in 1024 at least on terrain example)